oauthenticator/oauth2.py

```diff
@@ -86,23 +89,22 @@ def get_state(self):
             next_url = urlinfo._replace(
                 scheme="", netloc="", path="/" + urlinfo.path.lstrip("/")
             ).geturl()
-            if next_url != original_next_url:
-                self.log.warning(
-                    f"Ignoring next_url {original_next_url}, using {next_url}"
-                )
-        if self._state is None:
-            self._state = _serialize_state(
-                {"state_id": uuid.uuid4().hex, "next_url": next_url}
-            )
-        return self._state
+            return next_url
 
     def get(self):
         redirect_uri = self.authenticator.get_callback_url(self)
         token_params = self.authenticator.extra_authorize_params.copy()
         self.log.info(f"OAuth redirect: {redirect_uri}")
-        state = self.get_state()
-        self.set_state_cookie(state)
-        token_params["state"] = state
+
+        state_id = self._generate_state_id()
+        next_url = self._get_next_url()
```

Comment:
There was a `self._state` variable associated with the OAuthLoginHandler object that was read in `get_state` and reused if it was set previously.

After this PR, there is no such variable. Is this fine or a changed behavior? If each request to tornado leads to the creation of an entirely new OAuthLoginHandler object, I think its fine, and I think that is the case but I'm not sure.

Reply:
`self._state` is only an internal cache variable to avoid generating multiple UUIDs during one request. This shouldn't be an issue to lose.